### PR TITLE
Reader: improve accessibility of CommentsButton block

### PR DESCRIPTION
**Pull request by bluefuton**
#### Changes proposed in this Pull Request

* Make the comment button label readable by screenreaders.
* Use a `a` element for the component (currently uses a mixture of `div` and `button`).

Screen Shot 2021-02-04 at 15 27 31

#### Testing instructions

Go to Reader at http://calypso.localhost:3000 and find a post with a Reader comments button. Ensure that:

* Comments button is keyboard-accessible.
* Comments button has a blue focus outline when selected.

Screen Shot 2021-02-12 at 15 18 30

* When viewing a full post, the comments label is still hidden at mobile screen widths (should show count only).

Desktop width:

Screen Shot 2021-02-12 at 15 19 18

Mobile width:

Screen Shot 2021-02-12 at 15 19 23

Related to https://github.com/Automattic/wp-calypso/issues/46737.

### CODE REVIEW

File: client/blocks/comment-button/index.jsx [hunk 1]
```diff
@@ -20,51 +20,41 @@ import { getPostTotalCommentsCount } from 'calypso/state/comments/selectors';
...
+	return (
+		<a className="comment-button" href={ href } onClick={ onClick } target={ target }>
+			<Gridicon aria-hidden="true" icon="comment" size={ size } className="comment-button__icon" />
+			<span className="comment-button__label">
+				{ commentCount > 0 && (
+					<span className="comment-button__label-count">{ commentCount }</span>
+				) }
+				{ commentCount > 0 && (
+					<span className="comment-button__label-status">
+						{ translate( 'Comment', 'Comments', {
```
Comment (a8ci18n) [inline client/blocks/comment-button/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **164** times:
`translate( 'Comment', { context: 'noun'} )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/reader-comments-button-a11y)


:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

**Comment by bluefuton** [comment]
Todo:
- [x] make sure there's an href
- [x] make sure it's keyboard accessible
- [x] make sure focus styles work
- [x] insert a space for screen readers between count and label:

Screen Shot 2021-02-04 at 16 53 47

File: client/blocks/comment-button/index.jsx [hunk 2]
```diff
@@ -20,51 +19,50 @@ import { getPostTotalCommentsCount } from 'calypso/state/comments/selectors';
...
+					<span className="comment-button__label-count">{ commentCount }</span>
+				) }
 				<span className="comment-button__label-status">
-					{ translate( 'Comment', 'Comments', {
-						context: 'noun',
-						count: commentCount,
-					} ) }
+					{ commentCount > 0 && (
+						<Fragment>
+							{ translate( 'Comment', 'Comments', {
```
Comment (a8ci18n) [inline client/blocks/comment-button/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **164** times:
`translate( 'Comment', { context: 'noun'} )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/reader-comments-button-a11y)


:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

File: client/blocks/comment-button/index.jsx [hunk 3]
```diff
@@ -20,51 +19,57 @@ import { getPostTotalCommentsCount } from 'calypso/state/comments/selectors';
...
-			) }
-		</span>
+	return (
+		<a className="comment-button" href={ href } onClick={ onClick } tabIndex="0" target={ target }>
+			<Gridicon aria-hidden="true" icon="comment" size={ size } className="comment-button__icon" />
+			<span className="comment-button__label">
+				{ commentCount > 0 && (
+					<Fragment>
+						{ translate(
+							'{{count}}%(number)d{{/count}} {{label}}Comment{{/label}}',
```
Comment (a8ci18n) [inline client/blocks/comment-button/index.jsx:32]: :information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

File: client/blocks/comment-button/index.jsx [hunk 4]
```diff
@@ -20,51 +19,57 @@ import { getPostTotalCommentsCount } from 'calypso/state/comments/selectors';
...
-				<span className="comment-button__label-status">
-					{ translate( 'Comment', 'Comments', {
-						context: 'noun',
-						count: commentCount,
-					} ) }
-				</span>
-			) }
-		</span>
+	return (
+		<a className="comment-button" href={ href } onClick={ onClick } tabIndex="0" target={ target }>
```
Comment (jeryj) [inline client/blocks/comment-button/index.jsx:26]: A few things:
1. `tabIndex="0"` shouldn't be necessary if it's already a focusable element.
2. If we're locking this into being an `<a>`, then the naming of `comment-button` is misleading since it's not a button. 
3. Why are we locking this into being an `<a>`? Do we have any areas where we're using this comment button to reveal a comment section (not linked)?

File: client/blocks/comment-button/index.jsx [hunk 5]
```diff
@@ -20,51 +19,57 @@ import { getPostTotalCommentsCount } from 'calypso/state/comments/selectors';
...
+								},
+								context: 'noun',
+								count: commentCount,
+							}
+						) }
+					</Fragment>
+				) }
+				{ commentCount === 0 && (
+					<span className="comment-button__label-status">
+						>
```
Comment (jeryj) [inline client/blocks/comment-button/index.jsx]: Dangling `>` here.

Screen Shot 2021-02-12 at 9 13 52 AM

File: client/blocks/comment-button/index.jsx (hunk 3 above)
Comment (jeryj) [reply client/blocks/comment-button/index.jsx:32]: Along with this, will this line be able to handle reordering it to "Comments 4" if a language requires it? I _think_ it can since it's all within one translate string but I can't remember for sure.

File: client/blocks/comment-button/style.scss [hunk 6]
```diff
@@ -1,26 +1,29 @@
...
 	list-style-type: none;
 	padding: 4px;
 	position: relative;
 
-	&:hover,
-	&:active {
-		color: var( --color-primary );
+	.comment-button__label-status {
+		@include breakpoint-deprecated( '<480px' ) {
+			@include hide-content-accessibly();
```
Comment (jeryj) [inline client/blocks/comment-button/style.scss:12]: Nice! It's coming in handy in other places too :)

File: client/blocks/comment-button/style.scss [hunk 7]
```diff
@@ -1,26 +1,29 @@
...
 	display: inline-flex;
 	list-style-type: none;
 	padding: 4px;
 	position: relative;
 
-	&:hover,
-	&:active {
-		color: var( --color-primary );
+	.comment-button__label-status {
+		@include breakpoint-deprecated( '<480px' ) {
```
Comment (jeryj) [inline client/blocks/comment-button/style.scss:11]: Is there a non-deprecated one we should use instead?

File: client/blocks/reader-full-post/index.jsx [hunk 8]
```diff
@@ -551,7 +551,11 @@ export class FullPostView extends React.Component {
 								/>
 							) }
 
-							<div className="reader-full-post__comments-wrapper" ref={ this.commentsWrapper }>
+							<div
+								id="comments"
+								className="reader-full-post__comments-wrapper"
+								ref={ this.commentsWrapper }
```
Comment (jeryj) [inline client/blocks/reader-full-post/index.jsx:557]: It looks like this comments wrapper is used to scroll it into focus after a click on the comments link. It may be good to move focus to this element as well. At the moment it resets focus to the top of the page since it's a new page (expected behavior), but when someone clicks a comment(s) link, they would expect to be placed in a comments section. This could be a follow-up PR, as it may require a few more changes. Something like:

- Change from `div` to `section` to give it a tiny semantics boost
- Add a hidden "Comments" heading (`<h2>`? `<h3>`? Not sure which one) with an `aria-labelledby` referencing the `id` from the heading. This along with the `section` should get it to show-up in the landmarks menu. If not, then the heading is really useful anyways.
- Add a `tabindex="-1"` so it's programmatically focusable.
- Focus the element when it's available. Can likely place this somewhere around the `scrollTo` that's used.

File: client/blocks/reader-post-actions/index.jsx [hunk 9]
```diff
@@ -86,10 +87,10 @@ const ReaderPostActions = ( props ) => {
 			{ shouldShowComments( post ) && (
 				<li className="reader-post-actions__item">
 					<CommentButton
+						href={ getPostUrl( post ) + '#comments' }
 						key="comment-button"
 						commentCount={ post.discussion.comment_count }
 						onClick={ onCommentClick }
-						tagName="button"
```
Comment (jeryj) [inline client/blocks/reader-post-actions/index.jsx:92]: Why was this changed from a `button` to an `a`? Which one _should_ it be?

**Review by jeryj — CHANGES_REQUESTED** [review]
So much good stuff in here! 🙌🏻 I went through it fairly quickly, so don't take this as a full in-depth review. Thank you so much for taking this on!

File: client/blocks/reader-full-post/index.jsx (hunk 8 above)
Comment (jeryj) [reply client/blocks/reader-full-post/index.jsx:557]: I tested this with a screen reader, and focus is totally lost on the page change. It's like it's stuck "behind" on the old feed or something. This may be a deeper issue of always needing to move focus to the full page view after a click. We can at least fix this one though, and probably worth follow-up.

File: client/blocks/comment-button/index.jsx (hunk 3 above)
Comment (bluefuton) [reply client/blocks/comment-button/index.jsx:32]: Yes 👍 This wouldn't have worked previously, so it's an i18n improvement too :)

File: client/blocks/comment-button/style.scss (hunk 7 above)
Comment (bluefuton) [reply client/blocks/comment-button/style.scss:11]: The entire Reader layout is based on the old breakpoints, so for the moment I'm reluctant to dive into that particular rabbit hole 🐰

File: client/blocks/reader-post-actions/index.jsx (hunk 9 above)
Comment (bluefuton) [reply client/blocks/reader-post-actions/index.jsx:92]: This was actually prompted by your feedback on #46737:

> Comments: This should be a <a> link to the comments instead of a div along with the post title so it can be differentiated from other comment links, such as "[number] Comment(s): [post tile]


We'll always have an href to the comments section, even if we then employ extra functionality like smoothly scrolling to the right spot. Is `<a>` still the right element?

Comment (jeryj) [reply client/blocks/reader-post-actions/index.jsx:92]: haha - this sounds like me. I recently reviewed my own code not realizing it was my own, wondering why it had been done that way. 

As long as it's linking to a different spot, then an `<a>` is appropriate. If it were to _reveal_ a comments section, like "Show Comments" then a `<button>` may be more appropriate. That's what I'm concerned about with locking in the `a` - are there areas where we would use a button? If not, then an `<a>` is fine and we can refactor if needed later.
